pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+
+    def _merge_repositories(self, repo):
+        """
+        Move objects from the repo repository to the self repository
+        """
+        # TODO modify so it adds rather than replaces items, insure we
+        # use NocaseDict(This appears to be OK), and resolve
+        # conflicts during copy if there is data in our repository.
+        #
+        if repo.classes:
+            if not self.classes:
+                self.classes = repo.classes
+            else:
+                for ns in repo.classes:
+                    for cl in repo.classes[ns]:
+                        # test for existing qualifier
+                        self.classes[ns][cl.classname] = \
+                            repo.classes[ns][cl.classname]
+        if repo.instances:
+            if repo.instances:
+                if not self.instances:
+                    self.instances = repo.instances
+                else:
+                    for ns in repo.instances:
+                        for inst in repo.instances[ns]:
+                            # TODO test for duplicate instance names.
+                            self.instances[ns].extend(
+                                repo.instances[ns])
+        if repo.qualifiers:
+            if repo.qualifiers:
+                if not self.qualifiers:
+                    self.qualifiers = repo.qualifiers
+                else:
+                    for ns in repo.qualifiers:
+                        for ql in repo.qualifiers[ns]:
+                            # Test for existing qualifier
+                            self.qualifiers[ns][ql.name] = \
+                                repo.qualifiers[ns][ql.name]
+
+    def add_cimobjects(self, objects, namespace=None):
+        """
+            Add  a CIM object (CIMClass, CIMInstance, CIMQualifierDecl)
+            or list of objects to the faked repository.
+
+            This allows the user to create objects directly and insert them
+            into the repository for tests.
+        """
+        if not namespace:
+            namespace = DEFAULT_NAMESPACE
+        if isinstance(objects, list):
+            for obj in objects:
+                self.add_cimobjects(obj, namespace=namespace)
+
+            if self.verbose:
+                self.display_repository()
+
+        else:
+            obj = objects
+            if isinstance(obj, CIMClass):
+                cc = obj
+                if cc.superclass:
+                    try:
+                        _ = self.GetClass(cc.superclass,  # noqa: F841
+                                          namespace=namespace,
+                                          LocalOnly=True,
+                                          IncludeQualifiers=False)
+                    except CIMError as ce:
+                        if ce.args[0] == CIM_ERR_NOT_FOUND:
+                            ce.args = (CIM_ERR_INVALID_SUPERCLASS,
+                                       cc.superclass)
+                            raise
```

Comment:
I don't think that we should modify the exception we get and then re-raise. Plus, you don't know whether setting the the `args` attribute causes this to be visible in the other attributes such as `status_code`. (it actually is visible, but that is not documented, and we could change the implementation).

Reply:
Agreed. I think what we have is sufficient.  Just raised the new exception

Reply:
DONE.